src/openforms/appointments/admin.py

```diff
@@ -10,18 +13,37 @@
 from .fields import AppointmentBackendChoiceField
 from .models import AppointmentInfo, AppointmentsConfig
 from .registry import register
+from .utils import get_plugin
 
 
 @admin.register(AppointmentsConfig)
 class AppointmentsConfigAdmin(SingletonModelAdmin):
     def formfield_for_dbfield(self, db_field, request, **kwargs):
         if isinstance(db_field, AppointmentBackendChoiceField):
-            assert not db_field.choices
-            _old = db_field.choices
-            db_field.choices = register.get_choices()
-            field = super().formfield_for_dbfield(db_field, request, **kwargs)
-            db_field.choices = _old
-            return field
```

Comment:
this monkey patching suffers from race conditions, making a copy instead does not

Reply:
yeah but django's `ModelAdmin` instances are not thread safe and cause all these problems in the first place :( the alternative is re-implementing half of django's own admin formfield_for_dbfield, ModelForm fields from DB fields with custom admin widgets and a bunch of other things just to make the dynamic options be registered in a dropdown properly.

The Django admin code is an unfixable mess :(